marimbabot_planning/src/planning.cpp

```diff
@@ -215,7 +215,10 @@ moveit::planning_interface::MoveGroupInterface::Plan Planning::hit_note(
     
     // Concatinate trajectories
     auto plan = concatinated_plan({approach_plan, down_plan, retreat_plan});
-
+    
+    ros::Duration approach_duration = down_plan.trajectory_.joint_trajectory.points.back().time_from_start;
```

Comment:
Why do you only consider the time we need to approach the note, but note the time we need to hit it? This creates an error that adds up. Why do you sum up these durations either way? You only need to determine the time at which we reach the first note and add this to the relative timings provided in our goal. This gets weird if we consider clipping during the trajectory generation. If we include clipped timings here they will be used as ground truth and not considered as an issue, but if we don't all subsequent notes are slightly offset and get low scores. This being said, ideally we should never need to clip timings as they result in "wrong" but feasible timings. Ideally they should result in cancelation of the plan with an appropriate error code. Currently, the initial approach is always clipped, as we can not be at the first note in 0 seconds, but we would need to explicitly handle it in this case.